### PR TITLE
fix(Amount): correctly round the value upto 2 decimal places

### PR DESCRIPTION
**Pull request by snitin315**
Fixes https://github.com/razorpay/blade/issues/1572

### CODE REVIEW

File: packages/blade/src/components/Amount/__tests__/Amount.web.test.tsx [hunk 1]
```diff
@@ -132,5 +132,15 @@ describe('<Amount />', () => {
     expect(formatAmountWithSuffix({ value: 10000000, currency: 'MYR', suffix: 'none' })).toBe(
       '10,000,000',
     );
+    // Related issue - https://github.com/razorpay/blade/issues/1572
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Amount/__tests__/Amount.web.test.tsx:135]: add in native as well?

Comment (snitin315) [reply packages/blade/src/components/Amount/__tests__/Amount.web.test.tsx:135]: @kamleshchandnani Added

**Comment by snitin315** [comment]
I think this change is valid since we are passing the value as `123456.789`.

Screenshot 2023-08-30 at 12 07 56 PM

File: packages/blade/src/components/Amount/__tests__/Amount.native.test.tsx [hunk 2]
```diff
@@ -115,5 +115,15 @@ describe('<Amount />', () => {
...
     );
+    // Related issue - https://github.com/razorpay/blade/issues/1572
+    expect(formatAmountWithSuffix({ value: 2.07, currency: 'INR', suffix: 'decimals' })).toBe(
+      '2.07',
+    );
+    expect(formatAmountWithSuffix({ value: 2.077, currency: 'INR', suffix: 'decimals' })).toBe(
+      '2.08',
+    );
+    expect(formatAmountWithSuffix({ value: 2.3, currency: 'INR', suffix: 'decimals' })).toBe(
+      '2.30',
```
Comment (anuraghazra) [inline packages/blade/src/components/Amount/__tests__/Amount.native.test.tsx:126]: We can add few extra random edge cases also. 

0.2, 0.3, 100.1, 101.15, 2000.25 etc

**Review by anuraghazra — APPROVED** [review]

**Review by kamleshchandnani — APPROVED** [review]
